ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -537,6 +538,43 @@ type BiosConfigRef struct {
 // IronicInspect
 type IronicInspect string
 
+type SiteConfigMap struct {
+	Name      string            `yaml:"name"`
+	Namespace string            `yaml:"namespace"`
+	Data      map[string]string `yaml:"data"`
```

Comment:
@lack, that's actually a great point and something I struggled with quite a bit.
Because of the hub templating current implementation, we need to configure the values of the ConfigMap as strings, but we can still achieve multiline values in the ConfigMap as follows:

- The configmap:
> zone-1-cluster-log-fwd-outputs: "[{\"name\":\"kafka\", \"type\":\"kafka-open\", \"url\":\"tcp://10.46.55.190:9092/test\", \"secret\":{\"name\":\"kafka-open-secret\"}}]"

- The template:
> zone-1-cluster-log-fwd-outputs: "[{\"name\":\"kafka\", \"type\":\"kafka-open\", \"url\":\"tcp://10.46.55.190:9092/test\", \"secret\":{\"name\":\"kafka-open-secret\"}}]"

Links from the reference config:
- [configMap,](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/example/policygentemplates/hub-side-templating/group-du-sno-ranGen-templated.yaml#L25)
- [template](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/example/policygentemplates/hub-side-templating/site-du-sno-configmap.yaml#L27)

I hope this answers your concern.

Reply:
The ConfigMap API define [data](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L6869) as map[string]string . So it should be always map[string]string